cpp/src/Slice/Parser.cpp

```diff
@@ -5240,13 +5228,34 @@ Slice::Enum::recDependencies(set<ConstructedPtr>&)
     // An Enum does not have any dependencies.
 }
 
-Slice::Enum::Enum(const ContainerPtr& container, const string& name, const BuiltinPtr& underlying) :
+void
+Slice::Enum::initUnderlying(const TypePtr& type)
+{
+    assert(_contents.empty());
+
+    if (type)
```

Comment:
A comment to explain when type can be null could be good or we could fix the grammar to not call this method when not needed.

Reply:
I'll add a comment - I prefer to keep the code in `Grammar.y` small.